tests/unit/components/form/GitForm.spec.js

```diff
@@ -27,49 +60,94 @@ describe('Test component: GitForm', () => {
-      it('should be undefined on no git configuration in project', () => {
-        expect(wrapper.vm.repository).toBeUndefined();
-      });
-
-      it('should be not null on git configuration in project', () => {
-        saveProject({ id: 'test', git: { repository: 'testRepository' } });
+      it('should be undefined without any git configuration in project', () => {
         wrapper = shallowMount(GitForm, {
           props: {
-            projectName: 'test',
+            projectName: 'noGit',
           },
         });
+        expect(wrapper.vm.repository).toBeUndefined();
+      });
+
+      it('should not be null with git configuration in project', () => {
         expect(wrapper.vm.repository).toEqual('testRepository');
       });
     });
 
     describe('Test variable: token', () => {
-      it('should be undefined on no git configuration in project', () => {
+      it('should be undefined without any git configuration in project', () => {
+        wrapper = shallowMount(GitForm, {
+          props: {
+            projectName: 'noGit',
+          },
+        });
         expect(wrapper.vm.token).toBeUndefined();
       });
 
-      it('should be not null on git configuration in project', () => {
-        saveProject({ id: 'test', git: { token: 'testToken' } });
+      it('should not be null with git configuration in project', () => {
+        expect(wrapper.vm.token).toEqual('testToken');
+      });
+    });
+
+    describe('Test variable: username', () => {
+      it('should be undefined without any git configuration in project', () => {
         wrapper = shallowMount(GitForm, {
           props: {
-            projectName: 'test',
+            projectName: 'noGit',
           },
         });
-        expect(wrapper.vm.token).toEqual('testToken');
+        expect(wrapper.vm.username).toBeUndefined();
+      });
+
+      it('should not be null with git configuration in project', () => {
+        expect(wrapper.vm.username).toEqual('testUsername');
       });
     });
   });
 
   describe('Test function: onSubmit', () => {
-    it('should emit an event', () => {
+    it('should emit an event on success', async () => {
       wrapper.vm.repository = 'https://test/test.git';
-      wrapper.vm.token = 'tokenTest';
-      wrapper.vm.onSubmit();
+      wrapper.vm.token = 'testToken';
+      wrapper.vm.username = 'TestUsername';
 
-      expect(getProjectById('test').git).toEqual({
-        repository: 'https://test/test.git',
-        token: 'tokenTest',
+      let typeToCheck;
+      Notify.create = jest.fn(({ type }) => {
+        typeToCheck = type;
       });
+
+      await wrapper.vm.onSubmit();
+
+      expect(getProjectById).toBeCalled();
+      expect(saveProject).toBeCalled();
+      expect(updateGitProject).toBeCalled();
+      expect(wrapper.vm.submitting).toEqual(false);
       expect(wrapper.emitted()['project-git:save']).toBeTruthy();
+      expect(Notify.create).toBeCalled();
```

Comment:
Check the value Notify is called with, since multiple paths can lead to it's creation

Reply:
already done